### PR TITLE
feat(tokens): Add token refresh route

### PR DESCRIPTION
**Pull request by ian-fox**
* `PATCH` `/tokens/{id}/refresh` will refresh the token's value and return it while preserving the other fields of the token

Related to https://github.com/screwdriver-cd/screwdriver/issues/532

### CODE REVIEW

**Comment by tkyi** [comment]
I kind of like `renew` or `refresh`, but that's just semantics.

**Comment by ian-fox** [comment]
I don't have any attachment to the word `regenerate`. I'll ask at standup tomorrow.

File: test/plugins/tokens.test.js [hunk 1]
```diff
@@ -302,6 +302,69 @@ describe('token plugin test', () => {
...
+        });
+
+        it('returns 200 if regenerated successfully', () => {
+            const expected = Object.assign({}, testToken, { value: 'newValue' });
+
+            tokenMock.toJson.returns(Object.assign({}, tokenMock.toJson(), expected));
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 200);
+                assert.calledOnce(tokenMock.regenerate);
```
Comment (r3rastogi) [inline test/plugins/tokens.test.js]: Why are we using `regenerate` here but `refresh` elsewhere?

Comment (ian-fox) [inline test/plugins/tokens.test.js]: The function in the model is still called `regenerate`

I can put a PR in models to change that

Comment (ian-fox) [inline test/plugins/tokens.test.js]: https://github.com/screwdriver-cd/models/pull/168

**Pull request merged**